components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,724 @@
+		}
+	end
+end
+
+function CustomMatchGroupInput._processTeamPlayerMapData(players, opponentIndex, map, participants)
+	local tbdIndex = 0
+	local appendIndex = #players + 1
+
+	local playerIndex = 1
+	local playerKey = 't' .. opponentIndex .. 'p' .. playerIndex
+	while playerIndex <= MAX_NUM_PLAYERS_PER_MAP and (String.isNotEmpty(map[playerKey]) or
+		String.isNotEmpty(map[playerKey .. 'link']) or String.isNotEmpty(map[playerKey .. 'c'])) do
+
+		local player = map[playerKey .. 'link'] or map[playerKey]
+		if String.isEmpty(player) or Table.includes(TBD_PLAYERS, player:lower()) then
+			tbdIndex = tbdIndex + 1
+			player = TBD .. tbdIndex
+		else
+			-- allows fetching the link of the player from preset wiki vars
+			player = mw.ext.TeamLiquidIntegration.resolve_redirect(
+				map[playerKey .. 'link'] or Variables.varDefault(map[playerKey] .. '_page') or map[playerKey] or 'tbd'
+			)
+		end
+
+		local playerData = {
+			played = true,
+			cards = CustomMatchGroupInput._readCards(map[playerKey .. 'c']),
+		}
+
+		local match2playerIndex = CustomMatchGroupInput._fetchMatch2PlayerIndexOfPlayer(players, player)
+
+		-- if we have the player not present in match2player add basic data here
+		if not match2playerIndex then
+			match2playerIndex = appendIndex
+			playerData = Table.merge(playerData, {name = player, displayname = map[playerKey] or player})
+
+			appendIndex = appendIndex + 1
+		end
+
+		participants[opponentIndex .. '_' .. match2playerIndex] =  playerData
+
+		playerIndex = playerIndex + 1
+		playerKey = 't' .. opponentIndex .. 'p' .. playerIndex
+	end
+
+	return playerIndex - 1
+end
+
+function CustomMatchGroupInput._fetchMatch2PlayerIndexOfPlayer(players, player)
+	for match2playerIndex, match2player in pairs(players) do
+		if match2player and match2player.name == player then
+			return match2playerIndex
+		end
+	end
+end
+
+function CustomMatchGroupInput._readCards(input)
+	local cards = Json.parseIfString(input) or {}
+
+	for cardIndex, card in pairs(cards) do
+		if not CardNames[card:lower()] then
+			error('Invalid Card "' .. card .. '"')
+		end
+		cards[cardIndex] = CardNames[card]
+	end
+
+	return cards
+end
+
+-- function to sort out winner/placements
+function CustomMatchGroupInput._placementSortFunction(table, key1, key2)
+	local opponent1 = table[key1]
+	local opponent2 = table[key2]
+	local opponent1Norm = opponent1.status == 'S'
+	local opponent2Norm = opponent2.status == 'S'
+	if opponent1Norm then
+		if opponent2Norm then
+			return tonumber(opponent1.score) > tonumber(opponent2.score)
+		else return true end
+	else
+		if opponent2Norm then return false
+		elseif opponent1.status == 'W' then return true
+		elseif Table.includes(DEFAULT_LOSS_STATUSES, opponent1.status) then return false
+		elseif opponent2.status == 'W' then return false
+		elseif Table.includes(DEFAULT_LOSS_STATUSES, opponent2.status) then return true
+		else return true end
+	end
```

Comment:
```suggestion
	if opponent1Norm and opponent2Norm then
		return tonumber(opponent1.score) > tonumber(opponent2.score)
	elseif opponent1Norm then return true
	elseif opponent2Norm then return false
	elseif opponent1.status == 'W' then return true
	elseif Table.includes(DEFAULT_LOSS_STATUSES, opponent1.status) then return false
	elseif opponent2.status == 'W' then return false
	elseif Table.includes(DEFAULT_LOSS_STATUSES, opponent2.status) then return true
	else return true
	end
```
Should be logically the same,but I think somewhat easy to follow

Reply:
yeah way better